pandas/tseries/timedeltas.py

```diff
@@ -64,37 +66,30 @@ def to_timedelta(arg, unit='ns', box=True, errors='raise', coerce=None):
     """
     unit = _validate_timedelta_unit(unit)
 
-    def _convert_listlike(arg, box, unit, name=None):
-
-        if isinstance(arg, (list, tuple)) or not hasattr(arg, 'dtype'):
-            arg = np.array(list(arg), dtype='O')
-
-        # these are shortcutable
-        if is_timedelta64_dtype(arg):
-            value = arg.astype('timedelta64[ns]')
-        elif is_integer_dtype(arg):
-            value = arg.astype('timedelta64[{0}]'.format(
-                unit)).astype('timedelta64[ns]', copy=False)
-        else:
-            value = tslib.array_to_timedelta64(_ensure_object(arg),
-                                               unit=unit, errors=errors)
-            value = value.astype('timedelta64[ns]', copy=False)
-
-        if box:
-            from pandas import TimedeltaIndex
-            value = TimedeltaIndex(value, unit='ns', name=name)
-        return value
+    if errors not in ('ignore', 'raise', 'coerce'):
+        raise ValueError("errors must be one of 'ignore', "
+                         "'raise', or 'coerce'}")
 
     if arg is None:
         return arg
     elif isinstance(arg, ABCSeries):
         from pandas import Series
-        values = _convert_listlike(arg._values, box=False, unit=unit)
-        return Series(values, index=arg.index, name=arg.name, dtype='m8[ns]')
+        values = _convert_listlike(arg._values, unit=unit,
+                                   box=False, errors=errors)
+        try:
+            return Series(values, index=arg.index,
+                          name=arg.name, dtype='m8[ns]')
+        except ValueError:
```

Comment:
its confusing because you are doing it around the ``_convert_listlike``. This is saying that in some cases ``_convert_listlike`` can return invalid results and sometimes it will work depending on what you are *later* doing with it. This is a very bad pattern. All raising / catching should happen *inside* ``_convert_listlike``. This is the point of the ``box`` paramater. 

This makes it so much harder for a reader to understand what is going on.